tensorflow/python/kernel_tests/conv_ops_test.py

```diff
@@ -327,11 +327,28 @@ def testConv2DKernelSmallerThanStrideValid(self):
                        expected=expected_output)
 
   def testConv2DKernelSmallerThanStrideSame(self):
-    expected_output = [1, 3, 7, 9]
+    #[[1 4 7 0]
+    # [2 5 8 0]
+    # [3 6 9 0]
+    # [0 0 0 0]]
     self._VerifyValues(tensor_in_sizes=[1, 3, 3, 1],
                        filter_in_sizes=[1, 1, 1, 1],
                        strides=[2, 2], padding="SAME",
-                       expected=expected_output)
+                       expected=[1, 3, 7, 9])
+    # [[1  5  9 13]
+    #  [2  6 10 14]
+    #  [3  7 11 15]
+    #  [4  8 12 16]]
+    self._VerifyValues(tensor_in_sizes=[1, 4, 4, 1],
+                       filter_in_sizes=[1, 1, 1, 1],
+                       strides=[2, 2], padding="SAME",
+                       expected=[1, 3, 9, 11])
+
+    # TODO this currently fails.
+    #self._VerifyValues(tensor_in_sizes=[1, 8, 8, 1],
```

Comment:
Perhaps ignore that comment.  I'm just thinking about what's the right thing to do here, and I think instead of the change to use std::max(filter_size, stride_size), I think it might just be better to do real_padding = std::max(0, calculated_padding).

Why: the equations I usually go back to, from @Yangqing, master of understanding padding, is (for SAME):

R' = ceil(float(R) / float(S))
C' = ceil(float(C) / float(S))
Pr = ((R' - 1) * S + K - R) / 2
Pc = ((C' - 1) * S + K - C) / 2

In one case above (R=C=3, S=2, K=1), we have:  R' = C' = 2, Pr = Pc = 1 * 2 + 1 - 3 = 0, so it feels like the padding calculation is wrong for that case, since it's now calculated as '1' even though it's unneeded.

In the case that failed for you: R=C=4, S=2, K=1 we have R'=C'=2,  Pr = Pc = 1 * 2 + 1 - 4 = -1, so we have negative padding.  My guess is that you changed the code to avoid tripping the check failure -- if you use std::max(0, calculated_padding), I'm thinking the math will generally work out.  What do you think?